src/actions/index.js

```diff
@@ -7,6 +7,7 @@ const Actions = Reflux.createActions({
   validateConnectionString: { sync: true },
   onAuthSourceChanged: { sync: true },
   onAuthStrategyChanged: { sync: true },
+  onCancelConnectionAttemptClicked: { sync: true },
```

Comment:
Unrelated from PR .. In other plugins we have a mixture of sync true and false, i don't get what that's suppose to do though, what happens when is sync? it accumulate state changes and only emit them at the end?

Reply:
I think something in the store of `compass-connect` which is potentially a bug inducing practice is a combination of directly updating items on the state and later triggering a store update event `store.trigger(` with `store.setState`. I think we should only use one method, maybe `setState` to ensure we are performing the correct updates at the right time.

As for `sync: true` and `sync: false` it looks like it's a reflux pattern with triggering the store: https://github.com/reflux/refluxjs/tree/master/docs/actions#async-vs-sync-actions it's a bit confusing to me to be honest. How I understand it, all of these `sync` actions automatically cause a store `trigger` (and maybe more since some of these methods call trigger manually).